pkgs/top-level/all-packages.nix

```diff
@@ -5484,9 +5484,8 @@ with pkgs;
   };
 
   gpu-burn = callPackage ../applications/misc/gpu-burn {
-    # gpu-burn doesn't build on gcc11. CUDA 11.3 is the last version to use
-    # pre-gcc11, in particular gcc9.
-    stdenv = gcc9Stdenv;
+    cudaPackages = cudaPackages_11_8; # requires cuda 11.8
+    stdenv = gcc11Stdenv; # cuda 11.8 requires gcc 11
```

Comment:
Using `gcc11Stdenv` directly may cause libstdc++ mismatch errors (although unlikely here, gpu-burn being just an executable). For this reason there exists `cudaPackages_11_8.backendStdenv`, which is the same thing but uses libstdc++ compatible with the rest of nixpkgs. The name and details are subject to change, but generally we want to override the stdenv to use a compatible libstdc++